docs/plugins/testing_workshop_docs/article-3-readers-and-fixtures.md

```diff
@@ -0,0 +1,152 @@
+# Article 3: Readers and fixtures  
+
+This article explains how to use and test a plugin's reader function, built-in fixtures, custom fixtures, and enclosed testing. 
+
+## Article information  
+  
+* Article 1: [Python’s assert keyword](./article-1-pythons-assert-keyword.md) 
+* Article 2: [Pytest testing framework](./article-2-pytest-testing-frameworks.md)  
+* Article 3: This article (Readers and fixtures)  
+* Article 4: [Test coverage](./article-4-test-coverage.md)  
+* Resource links: [Testing resources](./testing-resources.md)  
+
+### This article covers:   
+* [Readers](#reader)  
+* [Built-in fixtures](#built-in-fixtures)  
+* [Custom fixtures and round-trip tests](#custom-fixtures-and-round-trip-tests)  
+* [Enclosed testing](#enclosed-testing)  
+  
+### Resources  
+The example plugin and all the tests discussed in this article are available in [this GitHub repository](https://github.com/DragaDoncila/plugin-tests).  
+  
+## Introduction  
+In this article, we discuss a plugin called `plugin_tests`, generated using the cookiecutter, which has a reader and a widget. The reader is the cookiecutter numpy file reader, `napari_get_reader`. It checks whether a path ends in `.npy`. If it doesn't, it returns `None`, and if it does, it returns the `reader_function`, which loads the data. 
+
+![napari_get_reader](../../images/napari_plugins_1st_napari_get_reader.PNG)
+  
+## Reader
+The `napari_get_reader` function is the first thing to test. In the top-level directory under `src`, we have the `plugin_tests` module. Inside `plugin_tests` is the `_tests` directory. This is a typical structure when writing tests. There is also a `test_reader.py` file, which is empty. We will populate it with tests.  
+
+![reader_function](../../images/napari_plugins_2nd_reader_function.png)
```

Comment:
This screenshot is missing what is returned from the function, which is quite important for the later discussion. Perhaps the docstring could be shortened to make room for this, and a new screenshot with the return values shown?

Reply:
Will research.

Reply:
Sean, I could not find a place where what is returned is displayed.  If you would send me a screenshot, I will be happy to include it. Thank you so much for all your contributions to this work.

Reply:
You are correct.  It would be much better to have a screenshot like that.

Reply:
Sean, we're going to publish without this screenshot.  We can add it later if we can find one. 
Thanks, 
Nadalyn